### PR TITLE
Internal components refactor [g3w-client #574]

### PR DESCRIPTION
**Pull request by Raruto**
## List of changes

1. internal JS/Vue components refactor: https://github.com/g3w-suite/g3w-client/pull/574
2. refactor search config API (ref: `WidgetSerializer` and `/api/config/<gid>/qdjango/<pid>`)
3. deprecate setting `G3W_CLIENT_SEARCH_ENDPOINT = 'ows'` (until removal, the new default is: `'api'`)

### CODE REVIEW

File: g3w-admin/qdjango/api/projects/serializers.py [hunk 1]
```diff
@@ -930,73 +927,63 @@ def __init__(self, instance=None, data=empty, **kwargs):
...
-                ret['options']['search_1n_relationid'] = body['relations']
-                del(body['relations'])
+                    'logicop': field.get('logicop', 'AND').upper(),
+                    'input': {
+                        **field['input'],
+                        'widget_type': etype(field, 'widgetv2type'),
+                        'type': ({
+                            'autocompletebox': 'autocompletefield',
+                            'selectbox': 'selectfield'
+                        }).get(field.get('widgettype'), field['input'].get('type')),
```
Comment (Raruto) [inline g3w-admin/qdjango/api/projects/serializers.py:972]: @wlorenzetti the `numberfield` type should also be returned in here (currently not implement in v3.7 response):


## OK: "Volume greater than" appears as number field

image

## KO: "Volume greater than" appears as text field

image

## Sample config: "Type and volume"

image

File: g3w-admin/base/settings/base.py [hunk 2]
```diff
@@ -291,6 +291,8 @@
 CLIENT_DEFAULT = 'client'
 CLIENT_G3WSUITE_LOGO = 'g3wsuite_logo_h40.png'
 CLIENT_OWS_METHOD = 'GET'
+
+# TODO: set qgis 'api' as default search endpoint?
 G3W_CLIENT_SEARCH_ENDPOINT = 'ows' #or 'api' for to use api layer vector with FieldFilterBackend
```
Comment (Raruto) [inline g3w-admin/base/settings/base.py]: @wlorenzetti some searches do not work with OWS + in the demo I see that is set as default:

```suggestion
G3W_CLIENT_SEARCH_ENDPOINT = 'api' # or 'ows' for to use api layer vector with FieldFilterBackend
```

image

Comment (Raruto) [reply g3w-admin/base/settings/base.py]: @wlorenzetti I think this option is deprecated, I get similar results on the stable branch (an empty select in that case):

> [`G3W_CLIENT_SEARCH_ENDPOINT`](https://g3w-suite.readthedocs.io/en/v3.7.x/settings.html#g3w-client-search-endpoint)
> 
> Search url endpoint for ‘searches calling’, default ows.
> - _ows:_ by wms search;
> - _api:_ by g3w-suite layer vector API.

## `G3W_CLIENT_SEARCH_ENDPOINT = 'ows'`

image

## `G3W_CLIENT_SEARCH_ENDPOINT = 'api'`

image

Comment (wlorenzetti) [reply g3w-admin/base/settings/base.py]: @raruto what type of search is it?

Comment (wlorenzetti) [reply g3w-admin/base/settings/base.py]: Ahh ok I read just now is a _Value Relation_ is it right?

Comment (Raruto) [reply g3w-admin/base/settings/base.py]: >  is a Value Relation is it right?

@wlorenzetti yes for the image above, but additionally (with the current branch) I think  you'll get: `No results found for this query/search` for almost all of them:

image

Comment (wlorenzetti) [reply g3w-admin/base/settings/base.py]: @volterra79 yes I get No results, but the response is correct, i.e. for _Type and sybtype_ search type: 
```
localhost:8006/ows/68/qdjango/336/?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetFeatureInfo&LAYERS=buildings_2f43dc1d_6725_42d2_a09b_dd446220104a&QUERY_LAYERS=buildings_2f43dc1d_6725_42d2_a09b_dd446220104a&INFO_FORMAT=application%2Fvnd.ogc.gml&FEATURE_COUNT=10000&CRS=EPSG%3A3857&I=0&J=0&FILTER=buildings_2f43dc1d_6725_42d2_a09b_dd446220104a%3A%22type%22+%3D+%27Administrative%27+AND+%22year%22+%3D+%272015%27&WITH_GEOMETRY=true
```

 I get a wms xml response :
```xml
<?xml version="1.0" encoding="UTF-8"?>
<wfs:FeatureCollection xmlns:qgs="http://qgis.org/gml" xmlns:ows="http://www.opengis.net/ows" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xlink="http://www.w3.org/1999/xlink" xsi:schemaLocation="http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.0.0/wfs.xsd http://qgis.org/gml" xmlns:wfs="http://www.opengis.net/wfs" xmlns:ogc="http://www.opengis.net/ogc" xmlns:gml="http://www.opengis.net/gml">
 <gml:featureMember>
  <qgs:buildings_2f43dc1d_6725_42d2_a09b_dd446220104a fid="buildings_2f43dc1d_6725_42d2_a09b_dd446220104a.48">
   <gml:boundedBy>
    <gml:Box srsName="EPSG:3857">
     <gml:coordinates cs="," ts=" ">1252098.43991029,5433342.30249331 1252122.43188242,5433364.29513444</gml:coordinates>
    </gml:Box>
   </gml:boundedBy>
   <qgs:geometry>
    <gml:MultiPolygon srsName="EPSG:3857">
     <gml:polygonMember>
      <gml:Polygon>
       <gml:outerBoundaryIs>
        <gml:LinearRing>
         <gml:coordinates cs="," ts=" ">1252098.43991029,5433348.8003191 1252102.97172725,5433364.29513444 1252122.43188242,5433352.53240365 1252119.49953027,5433342.30249331 1252098.43991029,5433348.8003191</gml:coordinates>
        </gml:LinearRing>
       </gml:outerBoundaryIs>
      </gml:Polygon>
     </gml:polygonMember>
    </gml:MultiPolygon>
   </qgs:geometry>
   <qgs:fid>48</qgs:fid>
   <qgs:name>aaaa</qgs:name>
   <qgs:address>Piazza Pietro Leopoldo</qgs:address>
   <qgs:year>2015</qgs:year>
   <qgs:photo>NULL</qgs:photo>
   <qgs:link>NULL</qgs:link>
   <qgs:form>NULL</qgs:form>
   <qgs:high>0</qgs:high>
   <qgs:volume>0</qgs:volume>
   <qgs:surface>153.5</qgs:surface>
   <qgs:architectural_barriers>()</qgs:architectural_barriers>
   <qgs:date_barriers>2024/03/29</qgs:date_barriers>
   <qgs:safety_exits>()</qgs:safety_exits>
   <qgs:date_exits>NULL</qgs:date_exits>
   <qgs:fire_system>()</qgs:fire_system>
   <qgs:date_fire>NULL</qgs:date_fire>
   <qgs:type>Administrative</qgs:type>
   <qgs:zone>B</qgs:zone>
   <qgs:attachment>false</qgs:attachment>
   <qgs:subtype>NULL</qgs:subtype>
   <qgs:user>NULL</qgs:user>
   <qgs:description>NULL</qgs:description>
  </qgs:buildings_2f43dc1d_6725_42d2_a09b_dd446220104a>
 </gml:featureMember>
</wfs:FeatureCollection>
```
but it is not show in g3w-client.

However I think the `G3W_CLIENT_SEARCH_ENDPOINT` option can be eliminated and default to 'ows' type.

Comment (Raruto) [reply g3w-admin/base/settings/base.py]: > However I think the G3W_CLIENT_SEARCH_ENDPOINT option can be eliminated and default to 'ows' type.

@wlorenzetti perhaps you meant `'api'` 😉 

> I get No results, but the response is correct

For those who can debug them, firefox support for `application/vnd.ogc.gml` is quite terrible 😞:

image

image

**Comment by Raruto** [comment]
@leolami https://github.com/g3w-suite/g3w-client/pull/574/commits/f99134b3cf75ca1c817065d4e63dd1756cf71017 takes care of empty query result for: `autompletefield` + `RelationReference`.

## Before

image


## After

image


---

### Unsearchable layer(s)

I think some check is missing on the server side.

From what I see, some other layers shouldn't be searchable either (unless relationships could break that option?):


image

**Comment by Raruto** [comment]
@wlorenzetti still waiting for your review → merge 😉

**Review by wlorenzetti — APPROVED** [review]
I seems works fine, I just make some tests.

**Pull request merged**